OpenRA.Mods.Common/AI/HackyAI.cs

```diff
@@ -450,12 +450,15 @@ bool HasAdequateAirUnitReloadBuildings(ActorInfo actorInfo)
 			if (aircraftInfo == null)
 				return true;
 
-			var ammoPoolsInfo = actorInfo.TraitInfos<AmmoPoolInfo>();
+			// We assume that UpgradeMinEnabledLevel = 0 means the actor reloads ammo by itself.
+			// To have ReloadAmmo also implies having at least 1 AmmoPool.
+			var buildingReload = actorInfo.TraitInfos<ReloadAmmoInfo>()
+				.Where(x => x.UpgradeMinEnabledLevel > 0).ToList();
```

Comment:
You could use `.FirstOrDefault()` instead of `ToList()` since you only care about the first result.